daterangepicker.js

```diff
@@ -1249,78 +1335,71 @@
+            if (!$(e.target).hasClass('available')) return;
+
+            var date = this.getDate(e);
 
             if (this.endDate || date.isBefore(this.startDate, 'day')) {
-                if (this.timePicker) {
-                    var hour = parseInt(this.container.find('.left .hourselect').val(), 10);
-                    if (!this.timePicker24Hour) {
-                        var ampm = this.container.find('.left .ampmselect').val();
-                        if (ampm === 'PM' && hour < 12)
-                            hour += 12;
-                        if (ampm === 'AM' && hour === 12)
-                            hour = 0;
-                    }
-                    var minute = parseInt(this.container.find('.left .minuteselect').val(), 10);
-                    var second = this.timePickerSeconds ? parseInt(this.container.find('.left .secondselect').val(), 10) : 0;
-                    date = date.clone().hour(hour).minute(minute).second(second);
-                }
-                this.endDate = null;
-                this.setStartDate(date.clone());
+
+                date = this.setTime('.left', date);
+
+                this.setDate(date.clone());
+
             } else if (!this.endDate && date.isBefore(this.startDate)) {
                 //special case: clicking the same date for start/end,
                 //but the time of the end date is before the start date
                 this.setEndDate(this.startDate.clone());
             } else {
-                if (this.timePicker) {
-                    var hour = parseInt(this.container.find('.right .hourselect').val(), 10);
-                    if (!this.timePicker24Hour) {
-                        var ampm = this.container.find('.right .ampmselect').val();
-                        if (ampm === 'PM' && hour < 12)
-                            hour += 12;
-                        if (ampm === 'AM' && hour === 12)
-                            hour = 0;
-                    }
-                    var minute = parseInt(this.container.find('.right .minuteselect').val(), 10);
-                    var second = this.timePickerSeconds ? parseInt(this.container.find('.right .secondselect').val(), 10) : 0;
-                    date = date.clone().hour(hour).minute(minute).second(second);
-                }
-                this.setEndDate(date.clone());
+                date = this.setTime('.right', date);
+
+                this.setDate(date.clone());
+
                 if (this.autoApply) {
                     this.calculateChosenLabel();
                     this.clickApply();
                 }
             }
 
             if (this.singleDatePicker) {
-                this.setEndDate(this.startDate);
+                this.setDate(date.clone());
+
                 if (!this.timePicker)
                     this.clickApply();
             }
 
-            this.updateView();
-
+            const onDateClick = true;
+            this.updateView(onDateClick);
```

Comment:
I was thinking about it. I thought it is more descriptive. I hate passing flags as parameters into function, because then you don't know what `true` stands for.